src/term-spec.js

```diff
@@ -128,21 +128,135 @@ declare export class AssignmentTargetPropertyProperty extends AssignmentTargetPr
 // class
 declare export class ClassExpression extends Expression {
   name?: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassDeclaration extends Statement {
+  isAbstract: any; // boolean
   name: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassElement extends Term {
-  isStatic: any;
+  isStatic: any; // boolean
+  accessModifier: any; // 'public' | 'protected' | 'private'
+}
+
+declare export class MethodClassElement extends ClassElement {
+  isAbstract: any; // boolean
+  hasQuestionToken: any; // boolean
   method: MethodDefinition;
 }
 
+declare export class ConstructorClassElement extends ClassElement {
+  params: ConstructorFormalParameters;
+  body: FunctionBody;
+}
+
+declare export class ConstructorFormalParameters extends Term {
+  items: ConstructorParameterDeclaration[];
+  rest?: ParameterDeclaration;
+}
+
+declare export class ConstructorParameterDeclaration extends Term {
+  accessModifier?: any; // 'public' | 'protected' | 'private'
+  hasReadonlyModifier: any; // boolean
+  parameter: ParameterDeclaration;
+}
+
+declare export class SemicolonClassElement extends ClassElement {}
```

Comment:
Yeah lets remove it. We of course have to parse them but as long as TypeScript doesn't give any semantic meaning to the extra semis there no reason to preserve them in the AST.